skupstina_django/scraper/scrape_utils.py

```
@@ -60,68 +60,40 @@ def parse_subject_details(url):
+        for act_period in list(periods_fd)[:2]:
             with open('scrapes_completed.txt', 'r+', encoding='utf8') as scrapes_completed:
                 if act_period not in scrapes_completed.read():
-                    parse_complete = False
                     act_period = act_period.strip()
                     print('\nScraping period: ', act_period)
                     url = (periods_url + act_period).replace(' ', '%20')
                     print(url)
+                    period_obj = write_period_to_db(act_period, url)
                     subjects, num_els = parse_subjects_list(url)
-                    all_subjects.extend(subjects)
-                    while not parse_complete:
-                        for subject in subjects:
-                            if 'details' not in subject:
-                                try:
-                                    print('Parsing ', subject['subject_url'])
-                                    subject_details = parse_subject_details(subject['subject_url'])
-                                    subject['details'] = subject_details
-                                    write_to_db(subject['details']['acts'])
-                                    print('Parsed  ', subject['subject_url'], ' **')
-                                    parse_complete = True
-                                except:
-                                    print('Error occurred in parsing ', subject['subject_url'])
-                                    parse_complete = False
+                    for subject in subjects:
+                        parse_complete = False
+                        print('Parsing ', subject['subject_url'])
+                        while not parse_complete:
+                            try:
+                                subject_details = parse_subject_details(subject['subject_url'])
+                                parse_complete = True
+                            except:
```

This is called a [Pokémon-exception](https://stackoverflow.com/questions/10594113/bad-idea-to-catch-all-exceptions-in-python) – "Gotta catch 'em all!" It's generally considered a bad idea. It's better to explicitly state which error are you trying to catch – even multiple ones if need be. 🙂 

Of course, there are some valid reasons – e.g. you want to make some code snippet super robust and ignore any errors. Maybe this is such a case – the parsing must continue no matter what.

---

Haha yeah. I'm not very fond of this either. I'll see if I can add some custom errors here.

---

Yeah. In general it's a good idea to give some limit to something that we want to run as a periodic task. Just to prevent the case where it really runs on and on for days and weeks and we have to manually intervene in the end.

Maybe something like:

```python
max_retries = 5
success = False
while not success and max_retries > 0:
    try:
        parse()
        success = True
    except:
        max_retries -= 1
```

---

Agreed! I've added one in